vkubelet/vkubelet.go

```diff
@@ -28,8 +25,8 @@ type Server struct {
 	k8sClient       *kubernetes.Clientset
 	taint           *corev1.Taint
 	provider        providers.Provider
-	podWatcher      watch.Interface
 	resourceManager *manager.ResourceManager
+	podCh           chan *corev1.Pod
```

Comment:
Just a thought, wdyt about passing a wrapped object that includes the a context that we can trace through from the original request?